### ISSUE TITLE
os.Getenv("HOME") is not populated on all editions of Windows

### ISSUE THREAD

**Issue by PaluMacil** [spf13/cobra#430]
I could not get my config file to pull in my name and license information. Upon investigation, I discovered that cobra/cmd/init.go adds a config path with `os.Getenv("HOME")`. However, despite this working on a Windows 10 Enterprise Edition (upgrade from Win 7, desktop) machine at my place of work, this line of code returns an empty string on my Windows 10 Professional Edition (OEM install, laptop) machine at home.

I determined that `os.Getenv("USERPROFILE")` will get my correct path to c:\user\username. Perhaps after checking HOME we could check for it being empty, and if it is empty, we could set it to USERPROFILE instead. If people seem open to that and would like me to make a pull request, I'll do it tomorrow or after whatever amount of discussion is needed.

The standard library (via `os/user`) appears to provide a more reliable but cgo-bound (let me know if you think I'm wrong--I'd love to use this if it weren't cgo-bound, and some things I read seem to indicate that the only limitation was Darwin, so I don't know if it was resolved) approach to get the home directory:
```
	user, err := usr.Current()
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(usr.HomeDir)
```

The last commit to touch this issue was here: https://github.com/spf13/cobra/commit/a3cd8ab85aeba3522b9b59242f3b86ddbc67f8bd

**Comment by PaluMacil** [comment]
Another approach would be to use https://github.com/mitchellh/go-homedir which is only about 117 sloc, going through a good variety of tests if previous tests fail. The license is MIT.

**Comment by bogem** [comment]
I think, `go-homedir` is good lib for this purpose. I will fix it in #428
